workflow/docs/design.md

```diff
@@ -0,0 +1,406 @@
+# Internal Design of Chaos Mesh Workflow
```

Comment:
I also think this file should be moved to another directory. 🙈

But I'm not sure about `cmd/workflow-engine`. Since the last talk with @YangKeao, we prefer workflow should be a part of "controller-manager", so we will not create another binary. How do you think about it?

And I have no idea where to place this file.

Reply:
I don't think so. Design doc lived in this dir should be fine, once it gets ready to be public, it would be better to move into [website repo](https://github.com/chaos-mesh/website/tree/master/docs), and each feature owner should take ownership, see an [example](https://github.com/kubernetes/website/pull/16018) of EndpointSlice feature from k8s website.